src/binder/bindertracing.cpp

```diff
@@ -81,31 +164,55 @@ bool BinderTracing::IsEnabled()
 
 namespace BinderTracing
 {
-    AssemblyBindOperation::AssemblyBindOperation(AssemblySpec *assemblySpec)
-        : m_bindRequest { assemblySpec }
-        , m_success { false }
+    AssemblyBindOperation::AssemblyBindOperation(AssemblySpec *assemblySpec, const WCHAR *assemblyPath)
+        : m_bindRequest { assemblySpec, nullptr, assemblyPath }
+        , m_populatedBindRequest { false }
+        , m_checkedIgnoreBind { false }
+        , m_ignoreBind { false }
+        , m_resultAssembly { nullptr }
         , m_cached { false }
     {
         _ASSERTE(assemblySpec != nullptr);
 
-        // ActivityTracker or EventSource may have triggered the system satellite load.
-        // Don't track system satellite binding to avoid potential infinite recursion.
-        m_trackingBind = BinderTracing::IsEnabled() && !m_bindRequest.AssemblySpec->IsMscorlibSatellite();
-        if (m_trackingBind)
-        {
-            m_bindRequest.AssemblySpec->GetFileOrDisplayName(ASM_DISPLAYF_VERSION | ASM_DISPLAYF_CULTURE | ASM_DISPLAYF_PUBLIC_KEY_TOKEN, m_bindRequest.AssemblyName);
-            FireAssemblyLoadStart(m_bindRequest);
-        }
+        if (!BinderTracing::IsEnabled() || ShouldIgnoreBind())
+            return;
+
+        PopulateBindRequest(m_bindRequest);
+        m_populatedBindRequest = true;
+        FireAssemblyLoadStart(m_bindRequest);
     }
 
     AssemblyBindOperation::~AssemblyBindOperation()
     {
-        if (m_trackingBind)
-            FireAssemblyLoadStop(m_bindRequest, m_success, m_resultName.GetUnicode(), m_resultPath.GetUnicode(), m_cached);
+        if (!BinderTracing::IsEnabled() || ShouldIgnoreBind())
+            return;
+
+        // Make sure the bind request is populated. Tracing may have been enabled mid-bind.
+        if (!m_populatedBindRequest)
+            PopulateBindRequest(m_bindRequest);
+
+        FireAssemblyLoadStop(m_bindRequest, m_resultAssembly, m_cached);
+    }
+
+    void AssemblyBindOperation::SetResult(PEAssembly *assembly, bool cached)
+    {
+        _ASSERTE(m_resultAssembly == nullptr);
+        m_resultAssembly = assembly;
+        if (m_resultAssembly != nullptr)
+            m_resultAssembly->AddRef();
```

Comment:
Shouldn't this ref be dropped in the dtor?

Reply:
`m_resultAssembly` is a `ReleaseHolder`, so it will call `Release` when it goes out of scope.